slick/src/main/scala/slick/util/InternalArrayQueue.scala

```diff
@@ -0,0 +1,175 @@
+package slick.util
+
+import java.util
+
+class InternalArrayQueue[E >: Null <: AnyRef](capacity: Int) {
```

Comment:
This class needs some documentation. Is it based on some JDK class with minor modifications? If so, what is the license or the original code?